airflow/models/dagbag.py

```diff
@@ -166,6 +166,31 @@ def dag_ids(self) -> List[str]:
         """
         return list(self.dags.keys())
 
+    @provide_session
+    def has_dag(self, dag_id: str, session: Session = None):
+        """
+        Checks the "local" cache, if it exists, and is not older than the configured cache time, return True
+        else check in the DB if the dag exists, return True, False otherwise
+        """
+        from airflow.models.serialized_dag import SerializedDagModel
+
+        if dag_id in self.dags:
+            min_serialized_dag_fetch_secs = timedelta(seconds=settings.MIN_SERIALIZED_DAG_FETCH_INTERVAL)
+            if (
+                dag_id in self.dags_last_fetched
+                and timezone.utcnow() > self.dags_last_fetched[dag_id] + min_serialized_dag_fetch_secs
+            ):
+                sd_last_updated_datetime = SerializedDagModel.get_last_updated_datetime(
+                    dag_id=dag_id,
+                    session=session,
+                )
+                if not sd_last_updated_datetime:
+                    return False
+                return not sd_last_updated_datetime > self.dags_last_fetched[dag_id]
```

Comment:
Why do we want this case?

This means if we have a sort of stale DAG it returns false.

Reply:
Oops, didn't know the sd means stale dag :)

Reply:
I see. No need. I will just return True